h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -732,7 +724,10 @@ private static NanoResponse maybeServeSpecial(RequestUri uri) {
       // url "/3/Foo/bar" => path ["", "GET", "Foo", "bar", "3"]
       String[] path = uri.getPath();
       if (path[2].equals("")) return redirectToFlow();
-      if (path[2].equals("Logs") && path[3].equals("download")) return downloadLogs();
+      if (path[2].equals("Logs") && path[3].equals("download")) {
+        LogArchiveContainer container = path.length >= 6 ? LogArchiveContainer.valueOf(path[4].toUpperCase()) : LogArchiveContainer.ZIP;
```

Comment:
This part could use some explanations. After looking into it, I think this is the part where we decide which type of `container` to use, but don't understand the logic.

What do you think of making this part ( maybe move it out to separate method) more self-explanatory?

Reply:
yeah, this is bad but that is because of how the whole method is designed :)

I will try to make it better, thx

Reply:
Yup, thanks a lot!